SeriesMaster-Frontend/App/Screens/DiscoverScreen.js

```diff
@@ -31,13 +41,24 @@ const DiscoverScreen = ({ navigation }) => {
   const [query, setQuery] = useState("");
   const [tvShows, setTvShows] = useState([]);
   const debouncedQuery = useDebounce(query, 500);
+  const responseListener = useRef();
 
   useEffect(() => {
+    registerForPushNotifications();
+
+    responseListener.current = Notifications.addNotificationResponseReceivedListener(
+      () => navigation.navigate("Profile")
+    );
+
+    fetchShows();
     async function fetchShows() {
       const shows = await getShows(debouncedQuery);
       setTvShows(shows);
     }
-    fetchShows();
+
+    return () => {
+      Notifications.removeNotificationSubscription(responseListener);
```

Comment:
If you don't accept comment above, I think it should be 
```suggestion
      Notifications.removeNotificationSubscription(responseListener.current);
```